vanilla-package/src/test/java/io/jenkins/jenkinsfile/runner/vanilla/SmokeTest.java

```diff
@@ -207,11 +243,17 @@ private String createTestRepoWithContentAndSCMConfigYAML(Map<String,String> file
             } catch (Exception e) {
                 throw new GitException("unable to write file", e);
             }
-            git.add(fileName);
+            // Stage this file.
+            git.add().addFilepattern(fileName).call();
         }
-        git.setAuthor(johnDoe);
-        git.setCommitter(johnDoe);
-        git.commit("Test commit");
+
+        // Commit the staged files.
+        git.commit()
+            .setAuthor(johnDoe)
+            .setCommitter(johnDoe)
+            .setMessage("Test commit")
+            .setSign(false) // Required if you're locally setup for signed commits.
```

Comment:
Locally, my gitconfig is setup for signed commits and that was this causing this test to fail with a `signing service is not available` exception message.  Using `org.jenkinsci.plugins.gitclient.GitClient`, it was not possible to configure it to disable signed requests.  Switching to `org.eclipse.jgit.api.Git` allows this an lets the test pass for folk with setups like mine.